gunicorn/glogging.py

```diff
@@ -313,18 +311,18 @@ def atoms(self, resp, req, environ, request_time):
         if hasattr(req_headers, "items"):
             req_headers = req_headers.items()
 
-        atoms.update({"{%s}i" % k.lower(): v for k, v in req_headers})
+        atoms.update({f"{{{k.lower()}}}i": v for k, v in req_headers})
```

Comment:
This (and two similar touchpoints within the next 20 lines) is one of the places where I wonder whether f-strings are an improvement or not.  The escaping logic is not completely intuitive.